### ISSUE TITLE
Disappointing Behavior with colors and Guide.manual_color_key

### ISSUE THREAD

**Issue by Nosferican** [GiovineItalia/Gadfly.jl#940]
Guide.manual_color_key renders an additional legend instead of replacing the default one from colors.
<img width="511" alt="screenshot 2016-12-16 14 47 21" src="https://cloud.githubusercontent.com/assets/2529329/21274470/c835233a-c39e-11e6-8546-a3d27abbc537.png">

**Comment by tlnagy** [comment]
Can you post a minimal code example for us to reproduce?

**Comment by Nosferican** [comment]
Here it is:
```
using DataFrames
using RDatasets
using Gadfly
iris = dataset("datasets", "iris")
```
Using `color` as usual:
```
Plot = plot(iris, x = :SepalWidth, y = :SepalLength, color = :Species, Geom.point)
```
Now I attempt to use `Guide.manual_color_key` along `color`.
```
Plot = plot(iris, x = :SepalWidth, y = :SepalLength, color = :Species, Geom.point,
Guide.manual_color_key("Not Species", ["setosa?", "versicolor?", "virginica?"],
["red", "green", "blue"]))
```
I suggest that `Guide.manual_color_key` replace the legend at all or add an option to either modify the one by `color` or replace it completely (the default behavior).

**Comment by Mattriks** [comment]
How about:

```julia
cscale = Scale.color_discrete_manual(colorant"red", colorant"green", colorant"blue")
p = plot(iris, x = :SepalWidth, y = :SepalLength, color = :Species, Geom.point, 
    cscale, Guide.colorkey("Iris")
)
```

See [Guide.manual_color_key](http://gadflyjl.org/stable/lib/guides/guide_manual_color_key.html) for an example of the sort of plot its designed for.

**Comment by Nosferican** [comment]
That's one alternative. I needed to change the label values as well to a string that it is not supported for column names in data frames so defining the `Scale.color_discrete_manual` would not be a complete solution for me, e.g. (p = 0.25). I ended up having to do the long way and make a layer for each one so I could use the `Guide.manual_color_key` and not depend on `color` to create the individual geoms for each group of y.

**Comment by Mattriks** [comment]
So the issue here seems to be what to do about difficult factor/level/label names in `DataFrames`. The suggestion here is to change the factor/level/label names after the dataframe has been converted to long format. For example:
```julia
d = Dict{String,String}("setosa"=>"p=0.1","versicolor"=>"p=0.2", "virginica"=>"p=0.3")
# Add a new variable to the dataframe:
iris[:g] = map(x -> d[x], iris[:Species])

p = plot(iris, x = :SepalWidth, y = :SepalLength, color = :g, Geom.point, 
    cscale, Guide.colorkey("Iris")
)
```

**Comment by Nosferican** [comment]
I like that approach as it is a good workaround and in many cases preferred to creating the different layers. However, I still think it is worthy to address the compatibility issue between `color` and `Scale.color_discrete_manual` and the double legend issue.

**Comment by xiaohk** [comment]
Yes, I agree with @Nosferican. It would be great if `Guide.manual_color_key` can overwrite the color key legend generated by `color` or `Scale.color_discrete_manual` .

**Comment by Nosferican** [comment]
Any updates on this front? An alternative would be to have the `Theme(key_position::Symbol = :right)` accept a vector of positions matching the number of legends. That way, one can potentially suppress any or display them at different positions.

On a potentially related aspect, for [Functions and Expressions](http://gadflyjl.org/stable/man/plotting.html#Functions-and-Expressions-1) the default `f1`, `f2`, ... names will almost surely be better served with more useful labels. I couldn't find an issue addressing it, but it might be related to this. In case there isn't a way to change the default names a custom color key might be needed and it would suffer from the same issue.

**Comment by Mattriks** [comment]
Not yet. Another possible option here is to give each `layer` control of its own key, like this (atm this has no effect):
```Julia
p = plot(
    layer([sin, cos], 0, 25, Theme(key_position=:none))
    )
```

**Comment by Mattriks** [comment]
Yes for [Functions and Expressions](http://gadflyjl.org/stable/man/plotting.html#Functions-and-Expressions-1), currently the function names are fixed [here](https://github.com/GiovineItalia/Gadfly.jl/blob/f106dfb2dcc992ac5eb7908557f2f1d961df21d1/src/statistics.jl#L1349-L1350).
